android/expoview/src/main/java/host/exp/exponent/experience/ExperienceActivity.java

```diff
@@ -124,32 +130,22 @@ public ExponentPackageDelegate getExponentPackageDelegate() {
   private DevBundleDownloadProgressListener mDevBundleDownloadProgressListener = new DevBundleDownloadProgressListener() {
     @Override
     public void onProgress(final @Nullable String status, final @Nullable Integer done, final @Nullable Integer total) {
-      UiThreadUtil.runOnUiThread(new Runnable() {
-        @Override
-        public void run() {
-          updateLoadingProgress(status, done, total);
+      UiThreadUtil.runOnUiThread(() -> {
+        if (!mIsLoading) {
+          showLoadingScreen(mManifest);
```

Comment:
would want to be sure that this doesn't cause any issues (e.g. loading screen popping up unexpectedly when it shouldn't), since the loading screen logic in this/related classes is somewhat brittle.

Reply:
Yeah, I totally agree on loading logic being very fragile and distributed across a few key classes, but what's been done here is just inlining previous function invocation. 😉
This PR does not change how the LoadingView-related logic at all.